src/plugins/data_source_management/public/components/utils.ts

```diff
@@ -89,19 +94,45 @@ export async function deleteDataSourceById(
   return savedObjectsClient.delete('data-source', id);
 }
 
+export async function deleteMultipleDataSources(
+  savedObjectsClient: SavedObjectsClientContract,
+  selectedDataSources: DataSourceTableItem[]
+) {
+  await Promise.all(
+    selectedDataSources.map(async (selectedDataSource) => {
+      await deleteDataSourceById(selectedDataSource.id, savedObjectsClient);
+    })
+  );
+}
+
+export async function createNewCredential(
+  savedObjectsClient: SavedObjectsClientContract,
+  newCredential: CreateNewCredentialType
+) {
+  return (
+    savedObjectsClient.create('credential', newCredential).then((response) => response.id || '') ||
+    null
+  );
+}
+
 export async function getExistingCredentials(savedObjectsClient: SavedObjectsClientContract) {
   const type: string = 'credential';
-  const fields: string[] = ['id', 'title'];
+  const fields: string[] = ['id', 'description', 'title', 'credentialMaterials'];
   const perPage: number = 10000;
```

Comment:
do we really need such a large number? we might need to test how it looks like with 10000 items

Reply:
Yes, we should test ... We can change the number. However, this is the default number we are using for all listing api calls